designer/client/src/components/graph/node-modal/node/NodeDetails.tsx

```diff
@@ -47,12 +47,11 @@ export function NodeDetails(props: NodeDetailsProps): JSX.Element {
     const dispatch = useDispatch();
 
     const performNodeEdit = useCallback(async () => {
-        await dispatch(editNode(scenario, node, applyIdFromFakeName(editedNode), outputEdges));
-
-        //TODO: without removing nodeId query param, the dialog after close, is opening again. It looks like props.close doesn't unmount component.
+        //TODO: without removing nodeId query param, the dialog after close, is opening again. It looks like useModalDetailsIfNeeded is fired after edit, because nodeId is still in the query string params, after scenario changes.
         mergeQuery(parseWindowsQueryParams({}, { nodeId: node.id }));
 
         props.close();
+        await dispatch(editNode(scenario, node, applyIdFromFakeName(editedNode), outputEdges));
```

Comment:
what if editNode fails for some reason? no time to display error, window already closed without saving.

Reply:
You are right, i changed the approach